conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/Lorre.scala

```diff
@@ -51,960 +25,45 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
-       * which would lead to inefficient storage performances as-is.
-       */
-      val saveBakers = (pages: LazyPages[tezosNodeOperator.DelegateFetchingResults]) =>
-        Source
-          .fromIterator(() => pages)
-          .mapAsync(1)(identity) //extracts the future value as an element of the stream
-          .mapConcat(identity) //concatenates the list of values as single-valued elements in the stream
-          .grouped(batchingConf.blockPageSize) //re-arranges the process batching
-          .mapAsync(1)(taggedBakers => {
-
-            val hashIds = taggedBakers.toList.map(_.blockHash.value)
-            val rolls = tezosNodeOperator.getRolls(hashIds)
-
-            rolls.map((taggedBakers, _))
-          })
-          .mapAsync(1) {
-            case (bakers, rolls) => processBakersPage(bakers, rolls)
-          }
-          .runFold(Monoid[Option[Int]].empty) { (processed, justDone) =>
-            processed |+| justDone
-          } andThen logOutcome
-
-      val fetchAndStore = for {
-        (bakerPages, _) <- prunedUpdates().map(tezosNodeOperator.getBakersForBlocks)
-        _ <- saveBakers(bakerPages) flatTap cleanup
-      } yield Done
-
-      fetchAndStore.transform(
-        identity,
-        e => {
-          val error = "I failed to fetch bakers from client and update them"
-          logger.error(error, e)
-          BakersProcessingFailed(message = error, e)
-        }
+    //displaying information for better debugging
+    displayInfo(indexer.platform.name, platformConf.network)
+    if (verbose.on)
+      displayConfiguration(
+        indexer.platform,
+        platformConf,
+        lorreConf,
+        (LORRE_FAILURE_IGNORE_VAR, sys.env.get(LORRE_FAILURE_IGNORE_VAR))
       )
 
-    }
-  }
-
-  /** Keeps track of time passed between different partial checkpoints of some entity processing
-    * Designed to be partially applied to set properties of the whole process once, and then only compute partial completion
-    *
-    * @param entityName a string that will be logged to identify what kind of resource is being processed
-    * @param totalToProcess how many entities there were in the first place
-    * @param processStartNanos a nano-time from jvm monotonic time, used to identify when the whole processing operation began
-    * @param processed how many entities were processed at the current checkpoint
-    */
-  private[this] def logProcessingProgress(entityName: String, totalToProcess: Int, processStartNanos: Long)(
-      processed: Int
-  ): Unit = {
-    val elapsed = System.nanoTime() - processStartNanos
-    val progress = processed.toDouble / totalToProcess
-    logger.info("================================== Progress Report ==================================")
-    logger.info("Completed processing {}% of total requested {}s", "%.2f".format(progress * 100), entityName)
+    //actual start of the indexer
+    indexer.start()
+  } finally shutdown()
```

Comment:
Why no shutdownHook?

What if the process is killed via OS signal, like a command line interrupt?
Will it shutdown cleanly?

Reply:
This is the same `try` we had: https://github.com/Cryptonomic/Conseil/blob/master/src/main/scala/tech/cryptonomic/conseil/Lorre.scala#L298 and this is what you are talking about: https://github.com/Cryptonomic/Conseil/pull/782/files/c8ab2b61d6f38a0789de939671fa13f03901fde4#diff-c7731ebc3651fb64cafd10746808698eR36 this is still there. I did not changed that :)

Reply:
You're right, I missed it hiding somewhere in the diffs 👍